cg/meta/create_validation_cases/validation_cases_api.py

```diff
@@ -0,0 +1,175 @@
+            fastq_files: list[File] = self.hk_api.get_files_from_latest_version(
+                bundle_name=validation_sample.from_sample, tags=[SequencingFileTag.FASTQ]
+            )
+            validation_bundle_path: Path = self.hk_api.get_latest_bundle_version(
+                bundle_name=validation_sample.internal_id
+            ).full_path
+            for fastq_file in fastq_files:
+                copy_file(file_path=Path(fastq_file.full_path), destination=validation_bundle_path)
+
+    def rename_and_add_original_sample_bundle_files(self, validation_samples: list[Sample]) -> None:
+        """Rename the fastq files to use the new validation sample internal id."""
+        for validation_sample in validation_samples:
+            validation_bundle_path: Path = self.hk_api.get_latest_bundle_version(
+                bundle_name=validation_sample.internal_id
+            ).full_path
+            fastq_files: list[Path] = get_files_matching_pattern(
+                directory=validation_bundle_path, pattern=SequencingFileTag.FASTQ
+            )
+            for fastq_file in fastq_files:
+                new_fast_file_path = Path(
+                    fastq_file.__str__().replace(validation_sample.from_sample),
+                    validation_sample.internal_id,
+                )
+                rename_file(file_path=fastq_file, renamed_file_path=new_fast_file_path)
+                new_tags: list[str] = self.get_new_tags(validation_sample)
+                self.add_validation_samples_to_hk(
+                    sample_id=validation_sample.internal_id,
+                    file_path=new_fast_file_path,
+                    tags=new_tags,
+                )
+
+    def get_new_tags(self, validation_sample: Sample) -> list[str]:
+        """
+        Get the fastq file tags from the original sample and
+        replace the bundle name with the validation sample internal id.
+        """
+        hk_files: list[File] = self.hk_api.get_files_from_latest_version(
+            bundle_name=validation_sample.from_sample, tags=[SequencingFileTag.FASTQ]
+        )
+        original_tags: list[str] = self.hk_api.get_tag_names_from_file(hk_files[0])
+        new_tags: list[str] = [validation_sample.internal_id]
+        for tag in original_tags:
+            if tag != validation_sample.from_sample:
+                new_tags.append(tag)
+        return new_tags
+
+    @staticmethod
+    def get_new_fastq_file_path(fastq_file: Path, validation_sample: Sample) -> Path:
+        return Path(
+            fastq_file.absolute().as_posix().replace(validation_sample.from_sample),
+            validation_sample.internal_id,
+        )
+
+    def add_validation_samples_to_hk(
+        self, sample_id: str, file_path: Path, tags: list[str]
+    ) -> None:
+        """Add the new validation sample bundle to housekeeoer."""
+        version: Version = self.hk_api.get_latest_bundle_version(sample_id)
+        self.hk_api.add_file(
```

Comment:
could we directly use `include_files_to_latest_version` here?

Reply:
include_files would attempt to copy the files into the bundle directory, since these are already there we don't need this. And I think it would also fail since the files it tries to copy already exist.